diff --git a/index.d.ts b/index.d.ts
index 938ba85aaf0e761ed47cf9b775ea42347c1f2155..8e8fd33e9c4038bde0a8b10140a894f4b97685b1 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -1,8 +1,11 @@
 import { BrowserifyObject, CustomOptions } from "browserify";
-import { CompilerOptions, ModuleKind, ScriptTarget } from "typescript";
+import typescript, { CompilerOptions, ModuleKind, ScriptTarget } from "typescript";

-export interface Options extends CustomOptions, CompilerOptions {
-	typescript?: string | import("typescript");
+// Provide local definition of Omit for compatibility with TypeScript <3.5
+type Omit<T, K extends keyof any> = Pick<T, Exclude<keyof T, K>>;
+
+export interface Options extends CustomOptions, Omit<CompilerOptions, "project"> {
+	typescript?: string | typeof typescript;
 	global?: boolean;
 	m?: ModuleKind;
 	p?: string | CompilerOptions;
@@ -10,6 +13,4 @@ export interface Options extends CustomOptions, CompilerOptions {
 	t?: ScriptTarget;
 }

-function tsify(b: BrowserifyObject, opts: Options): any;
-
-export = tsify;
+export default function tsify(b: BrowserifyObject, opts: Options): any;
